pkgs/tools/archivers/torrent7z/default.nix

```
@@ -0,0 +1,44 @@
+{ stdenv, fetchFromGitHub, fetchpatch, lib, ncurses }:
+
+stdenv.mkDerivation rec {
+  pname = "torrent7z";
+  version = "1.3";
+
+  src = fetchFromGitHub {
+    owner = "BubblesInTheTub";
+    repo = pname;
+    rev = "c982324b0af6094e7e28b106f50cb8bbec70d227";
+    sha256 = "+fSkB+u/CZ6VT+q8E0Ulvcr+BFVcxN4rCF4F0IyBFow=";
+  };
+
+  buildInputs = [ ncurses.dev ];
+
+  hardeningDisable = [ "format" ];
+
+  postPatch = ''
+    # Remove non-free RAR source code
+    # (see DOC/License.txt, https://fedoraproject.org/wiki/Licensing:Unrar)
+    rm -r linux_src/p7zip_4.65/CPP/7zip/Compress/Rar*
+    find . -name makefile'*' -exec sed -i '/Rar/d' {} +  
+  '';
+
+  preConfigure = ''
+    mkdir linux_src/p7zip_4.65/bin
+    cd linux_src/p7zip_4.65/CPP/7zip/Bundles/Alone
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ../../../../bin/t7z $out/bin
+  '';
+
+  meta = {
+    homepage = "https://github.com/BubblesInTheTub/torrent7z";
+    description = "A fork of torrent7z, viz a derivative of 7zip that produces invariant .7z archives for torrenting";
+    platforms = lib.platforms.linux;
+    maintainers = [ lib.maintainers.cirno-999 ];
+    mainProgram = "t7z";
+    # RAR code is under non-free UnRAR license, but we remove it
+    license = lib.licenses.gpl3Only;
+  };
```

```suggestion
  meta = with lib; {
    homepage = "https://github.com/BubblesInTheTub/torrent7z";
    description = "A fork of torrent7z, viz a derivative of 7zip that produces invariant .7z archives for torrenting";
    platforms = platforms.linux;
    maintainers = with maintainers; [ cirno-999 ];
    mainProgram = "t7z";
    # RAR code is under non-free UnRAR license, but we remove it
    license = licenses.gpl3Only;
  };
```
Avoids the repetition of lib.